### PR TITLE
feat: Proxies for server 4

### PR DESCRIPTION
**Pull request by ryanwohara**
# Description

Documenting proxy usage in a 4.x environment.

# Reasons

https://circleci.atlassian.net/browse/SERVER-1840

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc [hunk 1]
```diff
@@ -16,32 +16,47 @@ toc::[]
...
 * Some additional configuration is required to import orbs when installed behind a proxy. See https://circleci.com/docs/2.0/server-3-operator-orbs/#using-orbs-behind-a-proxy[Orbs on Server] docs for more information.
 * The JVM only accepts proxies that run over HTTP, not HTTPS, and therefore proxy URIs must be of the form `\http://user:password@host:port` rather than `\https://user:password@host:port`.
 * If your GitHub instance is running outside of the proxied environment (either GitHub.com or GitHub Enterprise), you must ensure that SSH traffic from our application (inside the Kubernetes cluster) and from our Nomad node can reach your instance without additional configuration. Our SSH agents do not respect proxy settings because it uses a different network protocol. If a proxy is the only way that any traffic can reach outside the proxied environment, this means it will block SSH traffic and your application will fail.
```
Comment (ryanwohara) [inline jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc:59]: This bullet point needs help. We cannot proxy ssh-keyscan but we can technically proxy SSH itself.

File: jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc [hunk 2]
```diff
@@ -16,32 +16,47 @@ toc::[]
...
+```yaml
+proxy:
+  enabled: "1"
+  http:
+    host: proxy.example.internal
+    port: "3128"
+    auth:
+      enabled: "1"
+      username: circleci
+      password: circleciforever
```
Comment (atulsingh0) [inline jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc:37]:
```suggestion
      username: <proxy-user>
      password: <proxy-password>
```

File: jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc [hunk 3]
```diff
@@ -16,32 +16,47 @@ toc::[]
...
+  no_proxy:
+    - localhost
+    - 127.0.0.1
+    - github.example.internal
+    - circleci.example.internal
+    - 10.0.0.0/16 # Subnets to exclude from the proxy (optional)
```
Comment (atulsingh0) [inline jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc:50]:
```suggestion
    - <vpc-or-subnet-cidr>   # VPC or Subnets to exclude from the proxy (optional)
```

**Review by lloydChris — CHANGES_REQUESTED** [review]
--We have given yaml snippets (instead of ex. 'proxy.https.host`) everywhere else, do we want to do that here?--

Both for consistency and so customers can copy paste?

File: jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc [hunk 4]
```diff
@@ -16,32 +16,47 @@ toc::[]
...
+    auth:
+      enabled: "1"
+      username: <proxy-user>
+      password: <proxy-password>
+  no_proxy:
+    - localhost
+    - 127.0.0.1
+    - github.example.internal
+    - circleci.example.internal
+    - <vpc-or-subnet-cidr>   # VPC or subnets to exclude from the proxy (optional)
```
Comment (lloydChris) [inline jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc:50]: Should we make a suggestion to put nomad subnets in here, or are those covered by something else?

File: jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc [hunk 5]
```diff
@@ -16,32 +16,47 @@ toc::[]
...
-* The load balancer endpoints must be added to the no-proxy list for the following services: `output processor` and `vm-service`. This is because the no-proxy list is shared between the application and build-agent. The application and build-agent are assumed to be behind the same firewall and therefore cannot have a proxy between them.
-* The KOTS Admin Console cannot be upgraded if proxy setting were configured. The proxy settings will be deleted and cause the KOTS Admin Console to break.
+* If your GitHub instance is running outside of the proxied environment (either GitHub.com or GitHub Enterprise), you must ensure that SSH traffic from our application (inside the Kubernetes cluster) and from our Nomad node can reach your instance. Please note the default `checkout` step in a CircleCI job will fail to clone code and our `ssh-keyscan` of GitHub Enterprise will not work. While you may configure an SSH proxy, `ssh-keyscan` can NOT be proxied and instead will require you provide `github.fingerprint` when using GHE.
```
Comment (lloydChris) [inline jekyll/_cci2/server/operator/installing-server-behind-a-proxy.adoc:59]: NIT:
```suggestion
* If your GitHub instance is running outside of the proxied environment (either GitHub.com or GitHub Enterprise), you must ensure that SSH traffic from CircleCI (inside the Kubernetes cluster) and from our Nomad node can reach your instance. Please note the default `checkout` step in a CircleCI job will fail to clone code and our `ssh-keyscan` of GitHub Enterprise will not work. While you may configure an SSH proxy, `ssh-keyscan` can NOT be proxied and instead will require you provide `github.fingerprint` when using GHE.
```

**Review by atulsingh0 — APPROVED** [review]

**Review by lloydChris — APPROVED** [review]
👍

**Comment by rosieyohannan** [comment]
@circleci/server where should we link to this guide from in the main installation guide? At which point do they need to decide this?

**Comment by ryanwohara** [comment]
@rosieyohannan Maybe under prerequisites? 🤔 
We need the proxy information to deploy the cluster so that makes sense to me

**Pull request merged**